### PR TITLE
STORM-2204 Adding Bolt side caching for HBase Lookup Bolt for 1.x branch

### PR DESCRIPTION
**Pull request by ambud**

### CODE REVIEW

**Comment by ambud** [comment]
https://github.com/apache/storm/pull/1783 original PR

**Comment by HeartSaVioR** [comment]
@ambud Code looks good, but indentation is really off. Could you take care of it?

File: external/storm-hbase/src/main/java/org/apache/storm/hbase/bolt/HBaseLookupBolt.java [hunk 1]
```diff
@@ -56,6 +66,30 @@ public HBaseLookupBolt withConfigKey(String configKey){
         this.configKey = configKey;
         return this;
     }
+    
+    @SuppressWarnings({ "unchecked", "rawtypes" })
+	@Override
+    public void prepare(Map map, TopologyContext topologyContext, OutputCollector collector) {
+    	super.prepare(map, topologyContext, collector);
+    	cacheEnabled = Boolean.parseBoolean(map.getOrDefault("hbase.cache.enable", "false").toString());
```
Comment (ambud) [inline external/storm-hbase/src/main/java/org/apache/storm/hbase/bolt/HBaseLookupBolt.java]: @HeartSaVioR sorry about that, I did see the PR build output on travis; have it on my TODO list.

**Comment by ambud** [comment]
@HeartSaVioR i fixed the issues; build failed because what seems to be a transient failure.

File: external/storm-hbase/pom.xml [hunk 2]
```diff
@@ -87,5 +87,9 @@
                 </exclusion>
             </exclusions>
         </dependency>
+        <dependency>
+        	<groupId>com.google.guava</groupId>
```
Comment (vesense) [inline external/storm-hbase/pom.xml]: Please use white space instead of tab space.

File: external/storm-hbase/pom.xml [hunk 3]
```diff
@@ -87,5 +87,9 @@
...
+        	<artifactId>guava</artifactId>
```
Comment (vesense) [inline external/storm-hbase/pom.xml]: same as above.

**Comment by ambud** [comment]
Done

**Comment by HeartSaVioR** [comment]
@ambud 
The last change looks good. Please squash the commits into one. I'll merge after rebasing.
If you mind squashing I'd be happy to do it instead.

**Comment by ambud** [comment]
@HeartSaVioR done, please let me know if any other changes need to be made. Looks like the space indentation is off my a little, please let me know if that will be an issue.

**Comment by vesense** [comment]
+1

**Comment by HeartSaVioR** [comment]
+1

**Comment by HeartSaVioR** [comment]
Master branch of storm-hbase still has guava as dependency even we use Caffeine, since hbase-server depends on Guava 16.0.1. 
So this patch doesn't add new guava dependency. I'll merge this.

**Pull request merged**